common/network-common/src/main/java/org/apache/spark/network/client/TransportClient.java

```diff
@@ -290,9 +290,11 @@ public void onFailure(Throwable e) {
     try {
       return result.get(timeoutMs, TimeUnit.MILLISECONDS);
     } catch (ExecutionException e) {
-      throw Throwables.propagate(e.getCause());
+      Throwables.throwIfUnchecked(e.getCause());
```

Comment:
It is unable to confirm what type of Throwable is wrapped in `ExecutionException`, so  still use

```
Throwables.throwIfUnchecked(e.getCause());
 throw new RuntimeException(e.getCause());
```